File: android/app/src/main/java/com/lightningkite/kwifttemplate/xml/DateRangeDemoXml.kt
//
// DateRangeDemoXml.swift
// Created by Kwift XML Android
//
package com.lightningkite.kwifttemplate.xml

import android.view.LayoutInflater
import android.view.View
import com.lightningkite.kwift.views.actual.ViewDependency
import com.lightningkite.kwifttemplate.R

class DateRangeDemoXml {

    lateinit var select: com.lightningkite.kwift.views.android.SelectDayView
    lateinit var range: com.lightningkite.kwift.views.android.SelectDateRangeView

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.date_range_demo, null, false)
        select = view.findViewById<com.lightningkite.kwift.views.android.SelectDayView>(R.id.select)
        range = view.findViewById<com.lightningkite.kwift.views.android.SelectDateRangeView>(R.id.range)
        return view
    }
}

File: iosOld/com/lightningkite/kwifttemplate/xml/layout/DateRangeDemoXml.swift
//
// DateRangeDemoXml.swift
// Created by Kwift XML
//

import UIKit
import FlexLayout
import PinLayout

class DateRangeDemoXml {
    
    weak var select: SelectDayView!
    weak var range: SelectDateRangeView!
    
    func setup(_ dependency: ViewDependency) -> UIView {
        return { () -> UIView in 
            let view = UIView(frame: .zero)
            view.layoutMargins = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
            view.flex.direction(.column).padding(0, 0, 0, 0).alignContent(.start).justifyContent(.start).define{ (flex) in 
                flex.addItem({ () -> SelectDayView in 
                    let view = SelectDayView(frame: .zero)
                    self.select = view
                    view.layoutMargins = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
                    return view
                }()
                ).margin(0, 0, 0, 0).grow(1).shrink(1).height(0).alignSelf(.stretch)
                
                flex.addItem({ () -> SelectDateRangeView in 
                    let view = SelectDateRangeView(frame: .zero)
                    self.range = view
                    view.layoutMargins = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
                    return view
                }()
                ).margin(0, 0, 0, 0).grow(1).shrink(1).height(0).alignSelf(.stretch)
                
            }
            return view
        }()
        
    }
}